x-pack/test/functional/page_objects/copy_saved_objects_to_space_page.ts

```diff
@@ -16,28 +17,27 @@ export function CopySavedObjectsToSpacePageProvider({
 }: FtrProviderContext) {
   const testSubjects = getService('testSubjects');
   const find = getService('find');
-  const { savedObjects } = getPageObjects(['savedObjects']);
+  const { savedObjects, common } = getPageObjects(['savedObjects', 'common']);
 
   return {
     async openCopyToSpaceFlyoutForObject(objectName: string) {
+      // This searchForObject narrows down the objects to those matching ANY of the words in the objectName.
+      // Hopefully the one we want is on the first page of results.
       await savedObjects.searchForObject(objectName);
+      await common.sleep(1000);
 
-      // Click action button to show context menu
-      await find.clickByCssSelector(
-        'table.euiTable tbody tr.euiTableRow td.euiTableRowCell:last-child .euiButtonIcon'
-      );
-
-      // Wait for context menu to render
-      await find.existsByCssSelector('.euiContextMenuPanel');
-
-      const actions = await find.allByCssSelector('.euiContextMenuItem');
-
-      for (const action of actions) {
-        const actionText = await action.getVisibleText();
-        if (actionText === 'Copy to space') {
-          await action.click();
-          break;
-        }
+      // this gets all the elements in the saved object table and keys them by title
+      const elements = keyBy(await savedObjects.getElementsInTable(), 'title');
+      // If there are more than 2 "actions" on the saved object we get the ... menu
+      // if there is the ... action menu we need to click it and click the copy saved object element
+      if (elements[objectName].menuElement) {
+        await elements[objectName].menuElement?.click();
+        // Wait for context menu to render
+        await find.existsByCssSelector('.euiContextMenuPanel');
+        await testSubjects.click('savedObjectsTableAction-copy_saved_objects_to_space');
+      } else {
+        // or there might be the copy saved object element without the menu
+        await elements[objectName].copySaveObjectsElement?.click();
```

Comment:
This same block of code is repeated in the `_import_objects` test.  I'm considering a refactor.

Reply:
I refactored it into the saved_objects_page